### PR TITLE
Lesson 9

### PR DESCRIPTION
**Pull request by namelexa**
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
+
+<style>
+    .button-homepage {
+        background: <?php echo $block->getData('button_color') ?>;
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: - styles need use in css file or use inline style
- use set/get methods like getButtonColor()

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <?php endif; ?>
+
+    <?php if ($block->getBannerContent()) : ?>
+        <p class='custom-widget-content'>
+            <?= $block->getBannerContent(); ?>
+        </p>
+    <?php endif; ?>
+<div class="home-banner">
+        <?php echo $block->getLayout()
+            ->createBlock('Magento\Cms\Block\Block')
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: This functionality need move in block class like function

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            ->setBlockId($block->getData('select_block'))
+            ->toHtml(); ?>
+    <?php if ($block->getData('button_title')): ?>
+        <button class="button-homepage" onclick="location.href='geekhub-cms'">
+            <?php echo $block->getData('button_title') ?>
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: - Use set/get magic method 
- onclick is bad decision, use <a class="button" href ....

**Review by APanteleychuk — CHANGES_REQUESTED** [review]
Need fix

File: app/code/AKisilenko/ModuleLesson9/etc/widget.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    </option>
+                    <option name="green"
+                            value="green"
+                            selected="true">
+                        <label translate="true">Green</label>
+                    </option>
+                </options>
+            </parameter>
+        </parameters>
+        <containers>
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/etc/widget.xml:46]: `<containers>  
<container..`
-  it's not needed

**Review by APanteleychuk — CHANGES_REQUESTED** [review]

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 5]
```diff
@@ -20,13 +10,11 @@
...
 <div class="home-banner">
-        <?php echo $block->getLayout()
-            ->createBlock('Magento\Cms\Block\Block')
-            ->setBlockId($block->getData('select_block'))
-            ->toHtml(); ?>
+        <?php echo $block->getHomeBanner(); ?>
     <?php if ($block->getData('button_title')): ?>
-        <button class="button-homepage" onclick="location.href='geekhub-cms'">
+        <a href="geekhub-cms" class="button-homepage"
+           style="background: <?php echo $block->getData('button_color') ?>;">
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: Use set/get magic method like $block->getButtonColor()

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 6]
```diff
@@ -20,13 +10,11 @@
...
     <?php endif; ?>
 <div class="home-banner">
-        <?php echo $block->getLayout()
-            ->createBlock('Magento\Cms\Block\Block')
-            ->setBlockId($block->getData('select_block'))
-            ->toHtml(); ?>
+        <?php echo $block->getHomeBanner(); ?>
     <?php if ($block->getData('button_title')): ?>
-        <button class="button-homepage" onclick="location.href='geekhub-cms'">
+        <a href="geekhub-cms" class="button-homepage"
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: href="geekhub-cms" - it's wrong, need use $block->getUrl('geekhub-cms')

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 7]
```diff
@@ -20,13 +10,11 @@
         </p>
     <?php endif; ?>
 <div class="home-banner">
-        <?php echo $block->getLayout()
-            ->createBlock('Magento\Cms\Block\Block')
-            ->setBlockId($block->getData('select_block'))
-            ->toHtml(); ?>
+        <?php echo $block->getHomeBanner(); ?>
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: no check if block no exist

**Review by APanteleychuk — CHANGES_REQUESTED** [review]

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 8]
```diff
@@ -9,12 +9,15 @@
...
-            <?php echo $block->getData('button_title') ?>
-        </a>
-    <?php endif; ?>
-</div>
\ No newline at end of file
+
+    <?php if ($block->getBannerContent()) : ?>
+        <div class="home-banner">
+                <?php echo $block->getHomeBanner(); ?>
+            <?php if ($block->getData('button_title')) : ?>
```
Comment (APanteleychuk) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: why you don't use set/get methods here and ather places?

**Review by APanteleychuk — CHANGES_REQUESTED** [review]

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml (hunk 8 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml]: all done

**Pull request merged**